launch_asc.ks

```diff
@@ -104,54 +97,32 @@ bays off.
 panels off.
 radiators off.
 
-
-if ship:status <> "PRELAUNCH" and stage:solidfuel = 0 {
-  // note that there's no SRB
-  set launch_tSrbSep to time:seconds.
-}
-
 lock steering to ascentSteering().
 lock throttle to ascentThrottle().
 
 /////////////////////////////////////////////////////////////////////////////
 // Enter ascent loop.
 /////////////////////////////////////////////////////////////////////////////
 
+local warped to false.
 until ship:obt:apoapsis >= apo {
-  ascentStaging().
-  ascentFairing().
-  wait launch_tick.
+	stagingCheck().
+	ascentDeploy().
+	if warped = 0 and altitude > min(ship:body:atm:height/10,1000) {
+		set warped to true.
+		physWarp(1).
+	}
+	wait 0.
 }
-
 uiBanner("Launch", "Engine cutoff").
 unlock throttle.
 set ship:control:pilotmainthrottle to 0.
-uiBanner("Launch", "Engine cutoff").
 
 /////////////////////////////////////////////////////////////////////////////
 // Coast to apoapsis and hand off to circularization program.
 /////////////////////////////////////////////////////////////////////////////
 
-// Get rid of ascent stage if less that 5% fuel remains ... bit wasteful, but
-// keeps our burn calculations from being erroneous due to staging mid-burn.
-if stage:resourceslex:haskey("LiquidFuel") {
-  if stage:resourceslex["LiquidFuel"]:capacity > 0 { // Checks to avoid NaN error
-    if stage:resourceslex["LiquidFuel"]:amount / stage:resourceslex["LiquidFuel"]:capacity < 0.05 {
-      uiBanner("ascend","Discarding ascent stage").
-      stage.
-      stagingPrepare.
-    }
-  }
-}
-// Corner case: circularization stage is not bottom most (i.e. there is an
-// aeroshell ejection in a lower stage).
-until ship:availablethrust > 0 {
-  stage.
-  uiBanner("ascend","Discard non-propulsive stage").
-  wait until stage:ready.
-}
-
-// Roll with top up.
+// Roll with top up
```

Comment:
It can cause problems if you start with boosters and control it with winglets, it does not have the power to turn 180 degrees. That is why I added the detection, not to do that if you face correctly already.

Reply:
It seems you were not talking about the roll for shuttles, but that line is about it. The original code always rotated the rocket (potentially shuttle) upside down and then 180 degrees here.

The other think is that I altered circ.ks that it points the face to future deltav vector instead of prograde.

Reply:
I was kind of annoyed by the initial 90° roll that was introduced to support shuttles; for rockets, it can cause stability issues during the critical first 2.5km of ascent. It sounds like you've fixed that, which is nice. (I do hope we're still somewhat compatible with shuttles -- for ascent rolling, it sounds like we are; my bigger concern is over utilFaceBurn. I will spend some time understanding that.)

Thank you for fixing my shitty `circ` logic when nodes unavailable! I just threw something together because I never play hardcore career mode; I always give myself enough initial funds for must-have upgrades.